pastis/matrix_generation/matrix_from_efields.py

```diff
@@ -95,7 +96,7 @@ def calculate_semi_analytic_pastis_from_efields(efields, efield_ref, direct_norm
     matrix_pastis_half = np.zeros([nb_modes, nb_modes])
 
     for pair in util.segment_pairs_non_repeating(nb_modes):
-        intensity_im = np.real((efields[pair[0]].electric_field - efield_ref) * np.conj(efields[pair[1]].electric_field - efield_ref))
+        intensity_im = np.real((efields[pair[0]] - efield_ref) * np.conj(efields[pair[1]] - efield_ref))
```

Comment:
@davidbourgeois the change you made in the return of `_luvoir_matrix_single_mode()` is exactly what was needed here to make it more streamlined 👍 This means that the entries of `efields` and `efield_ref` simply have to have the same type, but they're not tied to a specific type.

Now, does this work ok for RST, did you test this?

Reply:
@davidbourgeois confirmed that this works indeed

Reply:
I am @davidbourgeois and I confirm it.
----> OK I go out...